src/frontend/apps/impress/src/features/docs/doc-tree/hooks/useKeyboardActivation.ts

```diff
@@ -0,0 +1,30 @@
+import { useEffect } from 'react';
+
+import { SELECTORS } from '../components/dom-selectors';
+
+export const useKeyboardActivation = (
+  keys: string[],
+  enabled: boolean,
+  action: () => void,
+  capture = false,
+) => {
+  useEffect(() => {
+    if (!enabled) {
+      return;
+    }
+    const onKeyDown = (e: KeyboardEvent) => {
+      if (keys.includes(e.key)) {
+        e.preventDefault();
+        action();
+      }
+    };
+    const treeEl = document.querySelector<HTMLElement>('.c__tree-view');
```

Comment:
`.c__tree-view` could be a prop of this hook, it will then be a nice totally generic hook that you could reuse in different part of the app, that could live in `src/hook`, wdyt ?

```suggestion
    const elSelected = document.querySelector<HTMLElement>(selector);
```